FreeRTOS/FreeRTOS
# Migrate MSVC solution to VS2022

Description
-----------
- Add configuration to build for Win32 or x64.
- Minor code adjustments to fix compiler warnings.

Test Steps
-----------
- Load solution in Visual Studio. Build.
- Switch to x64 flavour. Build.

Checklist:
----------



Related Issue
-----------
- Discussed in [Forum](https://forums.freertos.org/t/wrong-types-in-x64-build-of-rtos-demo/18715/3).
- For successfully run x64 build, requires code from FreeRTOS/FreeRTOS-Kernel#924.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
